### PR TITLE
feat(orc8r): K8s Include labels and annotations based on the image/chart being deployed

### LINKED ISSUE #10103: Orc8r Pod Labels

### PR DESCRIPTION
**Pull request by joary**
## Summary

In essence the same as https://github.com/magma/magma/pull/9981, adding two points:

1. The labels also to pods/containers
2. The chart-version as a new label to both lte-orc8r and orc8r charts.

More details:

1. Include a label in the orc8r and lte-orc8r charts including the current version being deployed
    1. The label value comes directly from the docker image tag, which is the closes information the k8s have to the magma version
2. Include an annotation in the orc8r and lte-orc8r charts including the cluster release name being deploied (production/staging/green/blue)
    1. Annotation values comes from the helm chart name being deployed (configurable in terraform values)
3. OBS: For bot annotation and labels some K8s objects does not include this label since they’re independent of the magma version being run. Non exaustive list: Secrets/ConfigMaps/Certs/Roles/RoleBindings

Also fixed some bugs to allow testing:

1. Avoid to running metrics storeconfig job if prometheus is not enabled
2. Avoid creating PodDisruptionBudgets if not enabled

## Test Plan

A - Check the labels / annotations for each k8s object with helm template:

**Result for 'orc8r' helm chart:** https://gist.github.com/joary/e7156e6f6fce6834323aaa709925f6b8

**Result for 'lte-orc8r' helm chart:** 
https://gist.github.com/joary/aa3a8bf3974bc5ec9ca98ed35cd57325

B - Check the labels for each pod after deploying both orc8r and lte-orc8r charts

`kubectl get pods -n <namespace> --show-labels`:

https://gist.github.com/joary/c95906e247634f21f277d85772222c42

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by hcgatewood — COMMENTED** [review]
Looks like this is still stacked on the previous PR (?). Can you rebase it onto latest master then re-request a review?

File: orc8r/cloud/helm/orc8r/charts/nms/templates/nginx-proxy-deployment.yaml [hunk 1]
```diff
@@ -35,10 +35,14 @@ spec:
   selector:
     matchLabels:
 {{ tuple $envAll "nms" "nginx" | include "nms.selector-labels" | indent 6 }}
+{{ include "magmalte-image-version-label" . | indent 6 }}
   template:
     metadata:
       labels:
 {{ tuple $envAll "nms" "nginx" | include "nms.selector-labels" | indent 8 }}
+{{ include "magmalte-image-version-label" . | indent 8 }}
```
Comment (arunuke) [inline orc8r/cloud/helm/orc8r/charts/nms/templates/nginx-proxy-deployment.yaml]: Should we use magmalte-image-version-label for nginx-proxy deployment as well? Looking at our [artifactory for helm charts](https://artifactory.magmacore.org/artifactory/helm/), we have one for orc8r and one for lte-orc8r which I presume handles it for the controller and magmalte images respectively. If so, is there a possibility that nginx-proxy and magmalte could have different chart versions?

**Review by arunuke — CHANGES_REQUESTED** [review]
A general comment is to just check if all the pods we deploy are tagged with the image and chart versions as desired.

File: orc8r/cloud/helm/orc8r/charts/nms/templates/nginx-proxy-deployment.yaml (hunk 1 above)
Comment (joary) [reply orc8r/cloud/helm/orc8r/charts/nms/templates/nginx-proxy-deployment.yaml]: > Should we use magmalte-image-version-label for nginx-proxy deployment as well

No, I think we should use the `nginx-image-version-label`. I corrected the PR to reflect that.

> If so, is there a possibility that nginx-proxy and magmalte could have different chart versions?

No, both nginx deployments will use the same [helm value configuration](https://github.com/magma/magma/blob/master/orc8r/cloud/helm/orc8r/values.yaml#L105).

Being more clear, there are two nginx deployments: 
1. [One handling the public access to NMS](https://github.com/magma/magma/blob/master/orc8r/cloud/helm/orc8r/charts/nms/templates/nginx-proxy-deployment.yaml)
2. [Another handling the public access to Orc8r services](https://github.com/magma/magma/blob/master/orc8r/cloud/helm/orc8r/templates/nginx.deployment.yaml)

We expect them to have the same image-label value, though the char-version will be different, #1 will receive the NMS sub-chart version, and the #2 will receive Orc8r version.

**Review by arunuke — APPROVED** [review]
Thanks for the changes. LGTM.

I remember we had a plan to upgrade the EKS version as well (due to future support for nginx ingress). Is it going in a different PR?

**Comment by joary** [comment]
> I remember we had a plan to upgrade the EKS version as well (due to future support for nginx ingress). Is it going in a different PR?

Yes, I included in https://github.com/magma/magma/pull/10212, which is merged now. Rationale is that PR changes the k8s resources already, made more sense to be there.

**Review by hcgatewood — APPROVED** [review]
Skimmed, lgtm

**Pull request merged**